File: algorithms/BinaryResearch.ts
{
    let min:number = 0
    let max:number = 0
    let n:number = 0

    let inicialListNumber:number = 0
    let finalListNumber:number = 1000
    let randomN:number = 0
    let attempts:number = 0

    let sucess:boolean = false

    const start = () => {

        //start
        const inicial = performance.now();
        randomValue()
        n = finalListNumber/2
        if(randomN > n){ // prevent failure
            max = finalListNumber
        }
        attempt() // delete initial forecast

        //loop
        while(sucess === false){
            predicts()
            if( n === 0){
                break
            }
            attempts++
        }

        //sucess
        const end = performance.now();
        console.log("")
        console.log(`Numero aleatorio entre ${finalListNumber} e 0 selecionado: ${randomN}`)
        console.log(`O algoritimo levou ${end - inicial} ms`);
        console.log(`Numero de tentativas ${attempts}`);
        console.log("")

    }

    const randomValue = () => {

        randomN = Math.floor(Math.random() * (finalListNumber - inicialListNumber) + inicialListNumber)

    }

    const predicts = () => {
    
        n = (max + min) / 2
        Math.round(n)
        

        attempt()

    }

    const attempt = () => {

        if(n > randomN){
            max = n
        }else if(n < randomN){
            min = n
        }else{
            sucess = true
            console.log("")
            console.log("sucess")
            console.log("")
        }

    }

    start()
}

File: algorithms/BinaryResearch/BinaryResearch.py
from random import randint
import time

min = 0
max = 0
n = 0

inicialListNumber = 0
finalListNumber = 100000000
randomN = 0
attempts = 0

sucess = False

def start():
    
    global finalListNumber, n, randomN, max, attempts
    
    inicial = time.time()
    
    n = finalListNumber / 2
    randomValue()
    if(randomN > n):
        max = finalListNumber
    
    attempt()
    
    while(sucess == False):
        predicts()
        if( n == 0):
            break
        attempts = attempts + 1
    
    end = time.time()
    result = end - inicial
    print("")
    print("Numero aleatorio entre {0} e 0 selecionado: {1}".format(finalListNumber, randomN))
    print("O algoritimo levou {0}".format(result))
    print("Numero de tentativas {0}".format(attempts))
    print("")
    

def randomValue():
    
    global randomN
    
    randomN = randint(inicialListNumber, finalListNumber)
    
def predicts():
    
    global n, max, min
    
    n = (max + min) / 2
    n = round(n)
    
    attempt()
    
def attempt():
    
    global n, max, min, randomN, sucess
    
    if(n > randomN):
        max = n
    elif(n < randomN):
        min = n
    else:
        sucess = True
        print("")
        print("sucess")
        print("")
        
start()